src/harness/unittests/tsserverProjectSystem.ts

```diff
@@ -269,7 +263,7 @@ namespace ts {
         };
         readonly clearTimeout = (timeoutId: any): void => {
             if (typeof timeoutId === "number") {
-                this.callbackQueue.splice(timeoutId, 1);
+                unorderedRemoveItemAt(this.callbackQueue, timeoutId);

```

Comment:
This is called a "queue" but we never shift off the front; we just run all the callbacks with for-of. Still, it's possible that callers of `setTimeout` expect their callbacks to run in the order they were added. Should we change this to `removeItemAtPreservingOrder`?